packages/vite/src/client/client.ts

```diff
@@ -307,8 +307,16 @@ const enableOverlay = __HMR_ENABLE_OVERLAY__
 const hasDocument = 'document' in globalThis
 
 function createErrorOverlay(err: ErrorPayload['err']) {
+  const { customElements } = globalThis
+  let ErrorOverlayConstructor
+  if (customElements) {
+    ErrorOverlayConstructor = customElements.get(overlayId)
+  }
+  if (!ErrorOverlayConstructor) {
+    ErrorOverlayConstructor = ErrorOverlay
+  }
   clearErrorOverlay()
-  document.body.appendChild(new ErrorOverlay(err))
+  document.body.appendChild(new ErrorOverlayConstructor(err))
```

Comment:
I think we can simplify this code to:
```ts
  clearErrorOverlay()
  const { customElements } = globalThis
  if (customElements) {
    const ErrorOverlayConstructor = customElements.get(overlayId)!
    document.body.appendChild(new ErrorOverlayConstructor(err))
  }
```
When `createErrorOverlay` is called and custom elements is supported, `customElements.get(overlayId)` should return a value.

Reply:
Thanks for the review! I just push the [changes](https://github.com/vitejs/vite/pull/18647/commits/df14baa87cca9c983bb4033f98000ea458c3d86f).